apps/managedidentity/managedidentity.go

```diff
@@ -223,9 +235,44 @@ func (client Client) AcquireToken(ctx context.Context, resource string, options
 	if err != nil {
 		return base.AuthResult{}, err
 	}
+
+	fakeAuthInfo, err := authority.NewInfoFromAuthorityURI("https://login.microsoftonline.com/managed_identity", false, false)
+	if err != nil {
+		return base.AuthResult{}, err
+	}
+
+	fakeAuthParams := authority.NewAuthParams(client.miType.value(), fakeAuthInfo)
+	storageTokenResponse, err := cacheManager.Read(ctx, fakeAuthParams)
+	if err != nil {
+		return base.AuthResult{}, err
+	}
+	// ignore cached access tokens when given claims
+	if o.claims == "" {
+		ar, err := base.AuthResultFromStorage(storageTokenResponse)
+		if err == nil {
+			ar.AccessToken, err = fakeAuthParams.AuthnScheme.FormatAccessToken(ar.AccessToken)
+			return ar, err
+		}
+	}
 	tokenResponse, err := client.getTokenForRequest(req)
 	if err != nil {
 		return base.AuthResult{}, err
 	}
-	return base.NewAuthResult(tokenResponse, shared.Account{})
+	return client.AuthResultFromToken(ctx, fakeAuthParams, tokenResponse, true)
+}
+
+func (c Client) AuthResultFromToken(ctx context.Context, authParams authority.AuthParams, token accesstokens.TokenResponse, cacheWrite bool) (base.AuthResult, error) {
+	if !cacheWrite {
```

Comment:
When would `cacheWrite` be false?

Reply:
This is if we decide if we don't want to have cache for one method.
This is the active measure to have that. 
would you recommend we remove that ?

Reply:
The codepath with `cacheWrite=false` is never used by the product code, so it's essentially dead code. Let's not have dead code in the product.

Aside: MSAL always writes to cache.